### PR TITLE
support emu3-chat

### PR DESCRIPTION
**Pull request by mi804**
- [x] Document Updates
- [x] More Models or Datasets Support

### CODE REVIEW

File: swift/utils/module_mapping.py [hunk 1]
```diff
@@ -317,6 +317,8 @@ def __post_init__(self):
     connector='aligner',
     generator=['gen_vision_model', 'gen_aligner', 'gen_head', 'gen_embed'])
 
+EMU3_CHAT_KEYS = MultiModelKeys(language_model='model', )
```
Comment (Jintao-Huang) [inline swift/utils/module_mapping.py:320]: please check it

File: swift/llm/utils/template.py [hunk 2]
```diff
@@ -3176,6 +3177,61 @@ def data_collator(self, batch: List[Dict[str, Any]], padding_to: Optional[int] =
 register_template(TemplateType.molmo, MolmoTemplate(), lazy_tokenize=True, use_model=True)
 
 
+class Emu3ChatTemplate(Template):
+    system = 'You are a helpful assistant.'
+    image_placeholder = ['<|image token|>']
+
+    def __init__(self):
+        Template.__init__(self, [], [' User: {{QUERY}}. Assistant:'], ['<|extra_204|>'], ['<|extra_204|>'], self.system,
```
Comment (Jintao-Huang) [inline swift/llm/utils/template.py]: please check the user's previous ' '

File: swift/llm/utils/template.py [hunk 3]
```diff
@@ -3176,6 +3177,61 @@ def data_collator(self, batch: List[Dict[str, Any]], padding_to: Optional[int] =
...
+                          ['{{SYSTEM}}'])
+
+    def _encode(self, example: Dict[str, Any]) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+        inputs, _ = super()._encode(example)
+        if len(inputs) == 0:
+            return inputs, {}
+        inputs['input_ids'] = [self.tokenizer.bos_token_id] + inputs['input_ids']
```
Comment (Jintao-Huang) [inline swift/llm/utils/template.py]: please add bos_token in `Template.__init__(self, [],..)`

File: swift/llm/utils/template.py [hunk 4]
```diff
@@ -3176,6 +3177,61 @@ def data_collator(self, batch: List[Dict[str, Any]], padding_to: Optional[int] =
...
+        idxs = _findall(input_ids, self.tokenizer.encode(self.image_placeholder))
+        # Create image prompts
+        for i in range(len(raw_images)):
+            h, w = image_tokens[i].shape
+            imgstr = self.tokenizer.processor.to_imgstr(image_tokens[i])
+            image_prompt = (
+                self.tokenizer.boi_token + self.tokenizer.processor.prefix_template.format(H=h, W=w)
+                + self.tokenizer.img_token + imgstr + self.tokenizer.eol_token + self.tokenizer.eof_token
+                + self.tokenizer.eoi_token)
+            image_prompts.append(self.tokenizer.encode(image_prompt))
```
Comment (Jintao-Huang) [inline swift/llm/utils/template.py:3214]: please check `add_special_tokens=False` in encode function

File: swift/llm/utils/template.py [hunk 5]
```diff
@@ -3176,6 +3177,61 @@ def data_collator(self, batch: List[Dict[str, Any]], padding_to: Optional[int] =
...
+        added_tokens_len = 0
+        # Insert image tokens into input_ids
+        for idx, img_tokens in zip(idxs, image_prompts):
+            input_ids = input_ids[:idx + added_tokens_len] + img_tokens + input_ids[idx + added_tokens_len + 1:]
+            if labels is not None:
+                labels = labels[:idx + added_tokens_len] + [-100] * len(img_tokens) + labels[idx + added_tokens_len
+                                                                                             + 1:]
+            added_tokens_len += len(img_tokens) - 1
+
+        return {'input_ids': input_ids, 'labels': labels}
```
Comment (Jintao-Huang) [inline swift/llm/utils/template.py:3224]: please move post_encode code to encode

File: swift/llm/utils/template.py (hunk 4 above)
Comment (mi804) [reply swift/llm/utils/template.py:3214]: have checked, it's the same with offical implementation of emu3

File: swift/llm/utils/template.py (hunk 3 above)
Comment (mi804) [reply swift/llm/utils/template.py]: updated

**Review by Jintao-Huang — APPROVED** [review]

**Pull request merged**